lighthouse-core/gather/gatherers/trace.js

```diff
@@ -95,14 +99,21 @@ class Trace extends FRGatherer {
   /**
    * @param {LH.Gatherer.FRTransitionalContext} passContext
    */
-  async startSensitiveInstrumentation({driver}) {
+  async startSensitiveInstrumentation({driver, gatherMode}) {
     // TODO(FR-COMPAT): read additional trace categories from overall settings?
     // TODO(FR-COMPAT): check if CSS/DOM domains have been enabled in another session and warn?
     await driver.defaultSession.sendCommand('Page.enable');
     await driver.defaultSession.sendCommand('Tracing.start', {
       categories: Trace.getDefaultTraceCategories().join(','),
       options: 'sampling-frequency=10000', // 1000 is default and too slow.
     });
+
+    // Only inject the trace processor timespan marker, for timespan traces.
+    if (gatherMode !== 'timespan') return;
+
+    await driver.executionContext.evaluate(id => performance.mark(id), {
+      args: [TraceProcessor.TIMESPAN_MARKER_ID],
+    });
```

Comment:
surely there's _some_ better way of doing this, haha

Reply:
> what do you have in mind? :)

just expressing incredulity there's no `Tracing.insertEvent` or `Page.performanceMark()` or whatever :)

Reply:
you mean like [`Tracing.recordClockSyncMarker`](https://chromedevtools.github.io/devtools-protocol/tot/Tracing/#method-recordClockSyncMarker) 😄